common/templates/context_funcs.go

```diff
@@ -241,6 +242,67 @@ func targetUserID(input interface{}) int64 {
 	}
 }
 
+func (c *Context) tmplTargetHasRoleID(target interface{}, roleID interface{}) bool {
+	if c.IncreaseCheckCallCounter("has_role", 200) {
+		return false
+	}
+
+	targetID := targetUserID(target)
+	if targetID == 0 {
+		return false
+	}
+
+	ts, err := bot.GetMember(c.GS.ID, targetID)
+	if err != nil {
+		return false
+	}
+
+	role := ToInt64(roleID)
+	if role == 0 {
+		return false
+	}
+
+	c.GS.RLock()
```

Comment:
It's not needed to lock the guild state here, that's because bot.GetMember returns a copy from either the state or the api, not a direct reference, and while the role slice does point to the same array as the one in state, it's never modified (during MemberUpdate events the roles slice is instead completely replaced, meaning the copy would still hold the old role slice)

I notice i'm actually locking it myself in tmplHasRole but i think that was legacy code from when i directly referenced members from the state instead of copying them.

Reply:
Fantastic, I'll fix this right away!